File: WEEK 1/C++/Enhanced Dairy Farm Profit Calculation.cpp
#include <iostream>
using namespace std;

int main() {
    double milkCartonVolumeLiters, costPerLiter, profitPerCarton;
    
    cout << "Enter milk carton volume in liters: ";
    cin >> milkCartonVolumeLiters;

    cout << "Enter cost per liter: $";
    cin >> costPerLiter;

    cout << "Enter profit per carton: $";
    cin >> profitPerCarton;

    cout << "Enter the total amount of milk produced in the morning (liters): ";
    double totalMilkProduced;
    cin >> totalMilkProduced;

    int numberOfCartons = static_cast<int>(totalMilkProduced / milkCartonVolumeLiters);
    double costOfProduction = totalMilkProduced * costPerLiter;
    double profit = numberOfCartons * profitPerCarton;

    printf("Number of milk cartons needed: %d\n", numberOfCartons);
    printf("Cost of producing the milk: $%.2f\n", costOfProduction);
    printf("Profit for producing the milk: $%.2f\n", profit);

    return 0;
}

File: WEEK 1/C#/Enhanced Dairy Farm Profit Calculation.cs
using System;

class Program {
    static void Main() {
        Console.Write("Enter the total amount of milk produced in the morning (liters): ");
        decimal totalMilkProduced = decimal.Parse(Console.ReadLine());

        Console.Write("Enter the cost of producing one liter of milk: $");
        decimal costPerLiter = decimal.Parse(Console.ReadLine());

        Console.Write("Enter the profit per carton of milk: $");
        decimal profitPerCarton = decimal.Parse(Console.ReadLine());

        const decimal MilkCartonVolumeLiters = 3.78m;

        int numberOfCartons = (int)(totalMilkProduced / MilkCartonVolumeLiters);
        decimal costOfProduction = totalMilkProduced * costPerLiter;
        decimal profit = numberOfCartons * profitPerCarton;

        Console.WriteLine($"Number of milk cartons needed: {numberOfCartons}");
        Console.WriteLine($"Cost of producing the milk: R{costOfProduction:F2}");
        Console.WriteLine($"Profit for producing the milk: R{profit:F2}");
    }
}
